src/python-frontend/python_converter.cpp

```diff
@@ -251,10 +252,32 @@ exprt python_converter::get_expr(const nlohmann::json &element)
     if(element.contains("func") && element["_type"] == "Call")
     {
       std::string func_name = element["func"]["id"];
+
+      // Non-determinism functions restricted to basic types supported in Python
+      std::regex pattern(
+        R"(nondet_(int|char|bool|float)|__VERIFIER_nondet_(int|char|bool|float))");
+
+      if(std::regex_match(func_name, pattern))
+      {
+        // Function name pattern: nondet_(type). e.g: nondet_bool(), nondet_int()
+        size_t underscore_pos = func_name.rfind("_");
+        if(underscore_pos != std::string::npos)
+        {
+          std::string type = func_name.substr(underscore_pos + 1);
```

Comment:
This won't work for the `__VERIFIER_*` versions because func_name.find() will return the first match.

Reply:
If you were to write the regex like `(__VERIFIER_)?nondet_(int|char|bool|float)` you could then use the [`std::match_result::operator[]`](https://en.cppreference.com/w/cpp/regex/match_results/operator_at) on element 2, I think.

Reply:
Oh, sorry, you're using rfind. I missed that. Please ignore.